### PR TITLE
The EO Standard Library

### PR DESCRIPTION
**Pull request by nlchar**
Let's document the EO Standard Library and the language itself. 
A journey of a thousand miles begins with a single step... 
For now, the docs have the reference on `sprintf` and `stdout`.

### CODE REVIEW

**Comment by nlchar** [comment]
Added the `random` object reference.

**Comment by nlchar** [comment]
@yegor256 it would be great to have the wiki in the repo. However, the wiki does not have the option of pull requests. There are some [workarounds](https://stackoverflow.com/questions/10642928/how-to-pull-request-a-wiki-page-on-github) though. 
Egor, what do you think? 
There are three options:
1. We can place docs not in the wiki, but in the repo itself. 
2. You can grant me rights to write into the wiki repo.
3. We can set up one of the workarounds (quite hard for me, though).

**Comment by yegor256** [comment]
@nlchar I suggest we put this into our main README file

**Comment by nlchar** [comment]
> @nlchar I suggest we put this into our main README file
@yegor256 
there are gonna be too many docs for putting it into readme. :(
It's much better to divide them and put the link to the README. README should contain some basic examples and basic reference on the structure and features of EO, IMHO.

**Comment by yegor256** [comment]
@nlchar I think it's better to keep all important information in the README document, no matter how long it is. Here is an example: https://github.com/yegor256/takes

**Comment by yegor256** [comment]
@nlchar more formal and detailed descriptions of language features we'll publish in academic papers.

**Comment by nlchar** [comment]
@yegor256 okay. did it. see[ this repo ](https://github.com/nlchar/eo/tree/nlchar-docs)to have a look on the resulting README.

**Comment by nlchar** [comment]
@yegor256 I am still working on the reference on the eo library. I have some questions:
1. Could you review the current progress and tell your opinion? I'll add additional commits to the pull request and will tag you for the final review as the reference on lib is finished. I need the current review, though.
2. Should I describe `memory` and `while` at the moment or these objects are to be changed dramatically in the near future?
3. Should I describe `seq` object for sequencing computations or it will be removed or changed soon?

**Comment by yegor256** [comment]
@nlchar we have a lot of text already, but I'm not sure it's well structured. I'm sure it's long, but I'm not sure it's easy to read. Maybe we can define the structure first (table of contents)? Maybe we can decide what topics we cover and in which order?

**Comment by nlchar** [comment]
> @nlchar we have a lot of text already, but I'm not sure it's well structured. I'm sure it's long, but I'm not sure it's easy to read. Maybe we can define the structure first (table of contents)? Maybe we can decide what topics we cover and in which order?

I can add the ToC. There is also an automatic approach https://github.com/ekalinin/github-markdown-toc

**Comment by nlchar** [comment]
@yegor256 I've covered all the lib (except broken objects) plus added Table of Contents. Please, review the PR.

File: README.md [hunk 1]
```diff
@@ -264,10 +311,971 @@ This code will print this:
...
++package sandbox
++alias sprintf org.eolang.txt.sprintf
++alias stdout org.eolang.io.stdout
+
+[args...] > app
+  seq > @
+    stdout "Hello\n"
+    stdout "These objects\n"
+    stdout "will be datarized\n"
+    stdout "one by one, in sequatial order\n"
```
Comment (alex-semenyuk) [inline README.md:1235]: Typo: sequential

**Comment by yegor256** [comment]
@nlchar I think it's not a good idea to put this information into README. Moreover, I think it's a wrong idea to put this _anywhere_ aside from the source code itself. I suggest we implement the storage of objects first at objectionary.com and then make sure it can auto-generate documentation for any particular object. Make sense?

**Comment by nlchar** [comment]
> @nlchar I think it's not a good idea to put this information into README. Moreover, I think it's a wrong idea to put this _anywhere_ aside from the source code itself. I suggest we implement the storage of objects first at objectionary.com and then make sure it can auto-generate documentation for any particular object. Make sense?

Nice idea. However, it seems to be too challenging. Because the sources of the objects are scattered into `.eo` and `.java` files with ambiguous names. It took some investigation and reverse-composition of the code to write the docs. Hence, auto-generation seems very challenging for me as of now.

File: README.md (hunk 1 above)
Comment (nlchar) [reply README.md]: Thank you. Fixed that.

**Comment by nlchar** [comment]
> @nlchar I think it's not a good idea to put this information into README. Moreover, I think it's a wrong idea to put this _anywhere_ aside from the source code itself. I suggest we implement the storage of objects first at objectionary.com and then make sure it can auto-generate documentation for any particular object. Make sense?

Do you propose to reject the PR? The docs seem to be nice for this stage of the EO language. And I hope they would help the contributors and those who are interested in the project to see what can be done with the language as of now. 
Maybe we could save them (not in README, but in /docs folder) until the auto-generation solution is done? 
So, please, consider keeping it as an ad hoc solution. If we reject the docs, we will have zero knowledge of how to, for example, the `while` or `seq` actually works, etc.

**Comment by nlchar** [comment]
Moving to https://github.com/HSE-Eolang/eo

**Comment by 0crat** [comment]
Job `gh:cqfn/eo#185` is not assigned, can't get performer

**Pull request closed without merging**